spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index Page', type: :feature do
+  describe 'Viewing Post Index page' do
+    before(:each) do
+      @user1 = User.create(name: 'Tom', photo: 'https://unsplash.com/photos', bio: 'Teacher from Mexico.',
+                           posts_counter: 0)
+      @first_post = Post.create(author: @user1, title: 'post1', text: 'This is my first post', likes_counter: 0,
+                                comments_counter: 0)
+      @comment1 = Comment.create(post: @first_post, author: @user1, text: 'Hi Tom!, Nice comment')
+      visit user_posts_path(@user1) # Use the named route 'user_posts_path' to visit the posts index page for the user
+    end
+
+    it 'should show user name' do
+      expect(page).to have_content(@user1.name)
+    end
+
+    it 'should show user photo' do
+      expect(page.html).to include(@user1.photo)
+    end
```

Comment:
- [ ] The expectation `expect(page.html).to include(@user1.photo)` only checks if the HTML code includes the photo URL, but it doesn't ensure that the photo is actually displayed on the page. It would be better to use the have_selector matcher with an appropriate CSS selector to verify that the profile picture is visible. For example, you can use `expect(page).to have_selector("img[src='#{@user1.photo}']").`